### PR TITLE
Decode a Morse code message

### PR DESCRIPTION
**Pull request by TahenyBELGUITH**
In this milestone  we :  

- Create a method to decode a Morse code character, takes a string parameter, and return the corresponding character in uppercase .

- Create a method to decode an entire word in Morse code, takes a string parameter, and return the string representation. 
 
- Create a method to decode the entire message in Morse code, takes a string parameter, and return the string representation.

a code reviewer is appreciated 😄 
@TahenyBELGUITH @NickEmma

### CODE REVIEW

File: morse_message.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  new_str
+end
+
+def decode_result(str)
+  result = ''
+  str.split.each do |word|
+    result += decode_word(word)
+  end
+  result
+end
```
Comment (omarsalem7) [inline morse_message.rb:27]: - [ ] Kindly make sure when decoding a sentence for example when running `p decode_result(".-   -... --- -..-   ..-. ..- .-.. .-..    --- ..-.    .-. ..- -... .. . ...")`, I got this result `" ABOXFULLOFRUBIES"` kindly fix this issue by making a space to separate each word.
you can make something like the follwing:
```ruby
  str.split('  ').each do |word|
    result=result+ decode_word(word)
    result+=' '
  end
```

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi Team 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- There is no linter error present ✔️
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: morse_message.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (julie-ify) [inline morse_message.rb:27]: - [ ] The issue pointed out by the previous reviewer is still there. The sentence decode requires that you split the sentence using three spaces and decode each word, join those words and return it. Here is a suggestion

  ```
  def decode_result(str)
    result = ''
    str.split('   ').each do |word|
      result += decode_word(word)
      result += ' '
    end
    result.strip
  end
  ```
  What you have | What you should have
  :---:|:---:
  box | rub

**Review by julie-ify — CHANGES_REQUESTED** [review]
Hi @TahenyBELGUITH and @NickEmma,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TahenyBELGUITH** [comment]
Hi @julie-ify , thank you for your help 😄  we appreciate that!
this is what we have :  
image

**Review by gemmen29 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**